ui/src/serverIO.js

```diff
@@ -131,17 +131,16 @@ class ServerIO {
 
   // eslint-disable-next-line sonarjs/cognitive-complexity
   listen() {
+    this.refreshInterval = setInterval(sendRefreshSession, 9000);
```

Comment:
Not sure, perhaps you checked and tested but are we certain this is not called several times when navigating ?

Reply:
As long as any call to `listen()` has a corresponding clean-up call to `disconnect`, then it's fine I think. It seems to be the case here. That being said, I think `serverIO.js` definitely needs a broader review.

Reply:
:), yes !